plugins/plugin_utils/base.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+try:
+    import boto3
+    from botocore.client import Config
+except ImportError:
+    pass
+
+
+class AwsConnectionPluginBase:
+    def __init__(self) -> None:
+        pass
+
+    def _get_boto_client(
+        self,
+        service: str,
+        region_name: Optional[str] = None,
+        endpoint_url: Optional[str] = None,
+        config: Optional[Dict[str, Any]] = None,
+    ):
```

Comment:
Should the return type be `BaseClient` ?

Reply:
The botocore does use type hint, it is hard to deduced so putting `Any` as returned value is enough